apps/admin_api/lib/admin_api/v1/controllers/wallet_controller.ex

```diff
@@ -112,7 +111,7 @@ defmodule AdminAPI.V1.WalletController do
          {:ok, _} <- authorize(:enable_or_disable, conn.assigns, wallet),
          attrs <- Originator.set_in_attrs(attrs, conn.assigns),
          {:ok, updated} <- Wallet.enable_or_disable(wallet, attrs),
-         {:ok, updated} <- BalanceLoader.add_balances(updated) do
+         updated <- add_balances_if_allowed(updated, conn) do
```

Comment:
Won't this without `{:ok, updated}` let `{:error, code, description}` below slip into the happy path?